auth0/auth0-deploy-cli
# Supporting SCIM feature | Workaround on node-auth0@v3

### 🔧 Changes

In this PR, I'm adding a workaround to enable `SCIM` support on `auth0-deploy-cli` with the current `node-auth0@v3` version using the [Auth0 Management API](https://auth0.com/docs/api/management/v2/connections/get-scim-configuration). This lets us use `SCIM` features with the current SDK until we complete the migration to `node-auth0@v4`.

- New property `scim_configuration` is added to `connections`
- Modifiable  properties inside `scim_configuration`: `user_id_attribute`, `mapping`


### 📚 References
- [Auth0 Management API](https://auth0.com/docs/api/management/v2/connections/get-scim-configuration)



### 🔬 Testing

1. Create a `config.json` file and keep it in the root of your project. 
Example:
```
{
    "AUTH0_DOMAIN": "<YOUR_DOMAIN>",
    "AUTH0_CLIENT_ID": "<CLIENT_ID>",
    "AUTH0_CLIENT_SECRET": "<CLIENT_SECRET>",
    "AUTH0_INCLUDED_ONLY": ["themes"],
    "AUTH0_ALLOW_DELETE": false
}
```

2. Make sure you have SCIM supported enterprise connections configured on your tenant. The currently the SCIM supported strategies are `samlp`, `oidc`, `okta` and `waad`

3. Use the following commands to import or export configs:
Export YAML:
```node lib/index.js export -c config.json -o ./local -f yaml```
Import YAML:
 ```node lib/index.js import -c config.json --input_file ./local/tenant.yaml```
Refer documentation
4. Make changes to `scim_configuration`  inside `connections` and run `import` or `deploy` command. And observe the changes on your tenant.
5. Navigate to `Authentication > Enterprise > [YOUR_SCIM_CONNECTION] > Provisioning` and make changes to `SCIM` configuration. Run any `export` or `dump` command. Observe the changes on your local.
 



### 📝 Checklist
